scanner/src/funTest/kotlin/scanners/ScannerIntegrationFunTest.kt

```diff
@@ -100,6 +101,18 @@ class ScannerIntegrationFunTest : WordSpec({
                 patchExpectedResult(expectedResult)
         }
     }
+
+    "Scanning a project with the same provenance as packages" should {
+        val analyzerResult = createAnalyzerResultWithProject(project0, pkg0)
+        val ortResult = createScanner().scan(analyzerResult, skipExcluded = false, emptyMap())
+
+        "not have duplicated scan results" {
+            val expectedResult = readResource("/scanner-integration-shared-project-package-provenance.yml")
```

Comment:
Would it be sufficient (instead of asserting the expected result) to assert

1. `OrtResult.scanner.scanResults`has only one entry per `(provenance, scannerDetails)`.
2. `OrtResult.scanner.files` has only one entry per `provenance`.

I believe this what this commit ensures. Looking at these above proposed assertions, we could instead
add the as invariant checks into the constructor of `ScannerRun`.

Reply:
I now only test `ortResult.getScanResultsForId(project0.id).size shouldBe 1`, the more comprehensive tests are done in #10585.